apiserver/facades/client/client/filtering.go

```diff
@@ -220,6 +220,13 @@ func buildApplicationMatcherShims(a *state.Application, patterns ...string) (shi
 	}
 	shims = append(shims, unitShims...)
 
+	// Match on application status
+	statusInfo, err := a.Status()
```

Comment:
Need to think about this - it should probably be the derived display status rather than the raw app status. The derived status comes from the model cache app status.